### ISSUE TITLE
Support for non "off the record" profiles for modern web applications like whatsapp/google login and other

### ISSUE THREAD

**Issue by nicola-leoni** [emacs-eaf/emacs-application-framework#1135]
**Is your feature request related to a problem? Please describe.**
The new default profile for qt6-webengine is off the record, so each time the application is closed all the localStorage data is deleted. So all modern websites (eg: Whatsapp web, Google Docs, etc...) that stores data in the localStorage and local filesystem doesn't works and looses login/data every time.

**Describe the solution you'd like**
I fixed it configuring the qt6-webengine profile to be non off-the-record, I send the patch. Not a github expert, maybe consider to apply the patch in the repository.

```diff
diff --git a/core/webengine.py b/core/webengine.py
index 6a8ee4c..a8acde8 100755
--- a/core/webengine.py
+++ b/core/webengine.py
@@ -47,18 +47,20 @@ class BrowserView(QWebEngineView):
 
     translate_selected_text = QtCore.pyqtSignal(str)
 
-    def __init__(self, buffer_id):
-        super(QWebEngineView, self).__init__()
+    def __init__(self, profile, buffer_id):
+        super(QWebEngineView, self).__init__(profile)
 
         self.installEventFilter(self)
         self.buffer_id = buffer_id
         self.is_button_press = False
 
-        self.web_page = BrowserPage()
+        self.web_page = BrowserPage(profile)
         self.setPage(self.web_page)
 
         self.url_hovered = ""
         self.page().linkHovered.connect(self.link_hovered)
+        self.page().quotaRequested.connect(lambda request: request.accept())
+        self.page().fileSystemAccessRequested.connect(lambda request: request.accept())
 
         self.selectionChanged.connect(self.select_text_change)
 
@@ -781,8 +783,8 @@ Note, we need hook this function to signal 'loadProgress', signal 'loadStarted'
 
 
 class BrowserPage(QWebEnginePage):
-    def __init__(self):
-        QWebEnginePage.__init__(self)
+    def __init__(self, profile):
+        QWebEnginePage.__init__(self, profile)
 
     def execute_javascript(self, script_src):
         ''' Execute JavaScript.'''
@@ -824,6 +826,10 @@ class BrowserPage(QWebEnginePage):
         if self.url().toString() == "file:///":
             print("[JavaScript console]: " + message)
 
+webengine_profile = QWebEngineProfile('eaf')
+webengine_profile.setPersistentCookiesPolicy(QWebEngineProfile.PersistentCookiesPolicy.ForcePersistentCookies)
+webengine_profile.setHttpCacheType(QWebEngineProfile.HttpCacheType.DiskHttpCache)
+            
 class BrowserBuffer(Buffer):
 
     close_page = QtCore.pyqtSignal(str)
@@ -832,7 +838,9 @@ class BrowserBuffer(Buffer):
     def __init__(self, buffer_id, url, arguments, fit_to_view):
         Buffer.__init__(self, buffer_id, url, arguments, fit_to_view)
 
-        self.add_widget(BrowserView(buffer_id))
+        self.profile = webengine_profile
+
+        self.add_widget(BrowserView(self.profile, buffer_id))
 
         self.url = url
 
@@ -872,9 +880,7 @@ class BrowserBuffer(Buffer):
               "eaf-webengine-default-zoom",
               "eaf-webengine-zoom-step"])
 
-        # The default profile can be accessed by defaultProfile(). It is a built-in profile that all web pages not specifically created with another profile belong to.
-        self.profile = QWebEngineProfile.defaultProfile()
-        self.profile.defaultProfile().setHttpUserAgent(self.pc_user_agent)
+        self.profile.setHttpUserAgent(self.pc_user_agent)
 
         self.caret_js_ready = False
         self.caret_browsing_mode = False
@@ -892,7 +898,7 @@ class BrowserBuffer(Buffer):
         self.buffer_widget.web_page.fullScreenRequested.connect(self.handle_fullscreen_request)
         self.buffer_widget.web_page.pdfPrintingFinished.connect(self.notify_print_message)
         self.buffer_widget.web_page.featurePermissionRequested.connect(self.permission_requested) # enable camera permission
-        self.profile.defaultProfile().downloadRequested.connect(self.handle_download_request)
+        self.profile.downloadRequested.connect(self.handle_download_request)
 
         self.settings = self.buffer_widget.settings()
         try:
@@ -917,6 +923,7 @@ class BrowserBuffer(Buffer):
             self.settings.setFontSize(QWebEngineSettings.FontSize.DefaultFontSize, self.font_size)
             self.settings.setFontSize(QWebEngineSettings.FontSize.DefaultFixedFontSize, self.fixed_font_size)
 
+            self.settings.setAttribute(QWebEngineSettings.WebAttribute.LocalStorageEnabled, True)
             self.settings.setAttribute(QWebEngineSettings.WebAttribute.FullScreenSupportEnabled, True)
             self.settings.setAttribute(QWebEngineSettings.WebAttribute.DnsPrefetchEnabled, True)
             self.settings.setAttribute(QWebEngineSettings.WebAttribute.FocusOnNavigationEnabled, True)
@@ -1488,12 +1495,12 @@ class BrowserBuffer(Buffer):
     @interactive(insert_or_do=True)
     def toggle_device(self):
         ''' Toggle device.'''
-        user_agent = self.profile.defaultProfile().httpUserAgent()
+        user_agent = self.profile.httpUserAgent()
         if user_agent == self.pc_user_agent:
-            self.profile.defaultProfile().setHttpUserAgent(self.phone_user_agent)
+            self.profile.setHttpUserAgent(self.phone_user_agent)
             self.set_aspect_ratio(2.0 / 3)
         else:
-            self.profile.defaultProfile().setHttpUserAgent(self.pc_user_agent)
+            self.profile.setHttpUserAgent(self.pc_user_agent)
             self.set_aspect_ratio(0)
 
         self.refresh_page()
@@ -1641,11 +1648,11 @@ class BrowserBuffer(Buffer):
         self.change_url(url)
 
     def set_agent(self, agent):
-        user_agent = self.profile.defaultProfile().httpUserAgent()
+        user_agent = self.profile.httpUserAgent()
         if agent == "pc" and user_agent == self.phone_user_agent:
-            self.profile.defaultProfile().setHttpUserAgent(self.pc_user_agent)
+            self.profile.setHttpUserAgent(self.pc_user_agent)
         elif agent == "phone" and user_agent == self.pc_user_agent:
-            self.profile.defaultProfile().setHttpUserAgent(self.phone_user_agent)
+            self.profile.setHttpUserAgent(self.phone_user_agent)
 
     @PostGui()
     def _change_url(self, url):
@@ -1721,9 +1728,6 @@ class CookiesManager(object):
 
         self.cookies_dir = os.path.join(get_emacs_config_dir(), "browser", "cookies")
 
-        # Both session and persistent cookies are stored in memory
-        self.browser_view.page().profile().setPersistentCookiesPolicy(QWebEngineProfile.PersistentCookiesPolicy.NoPersistentCookies)
-
         self.cookie_store = self.browser_view.page().profile().cookieStore()
 
         self.cookie_store.cookieAdded.connect(self.add_cookie)      # save cookie to disk when captured cookieAdded signal

```

**Comment by MatthewZMD** [comment]
@manateelazycat Does it look good to you, if so I can merge the patch.

**Comment by manateelazycat** [comment]
> @manateelazycat Does it look good to you, if so I can merge the patch.

This patch looks good. I have a fever recently. Thanks for the help in merging it.

**Comment by nicola-leoni** [comment]
I've also removed the class CookiesManager because now is not more necessary, leaving the cookies handling to Chromium. This fixes the issue for apps like Teams web that tries to save cookies with a key longer of 255 bytes,  which gives error because CookiesManager uses a filename with the cookie key. But this breaks the cookies importing from Chrome, which I don't use (I never manage to make it work). I'm not sure if you want to patch it.

**Comment by MatthewZMD** [comment]
> This patch looks good. I have a fever recently. Thanks for the help in merging it.

Oh no, take care!

> I've also removed the class CookiesManager

At least in this patch, I don't see where the CookiesManager is removed though.

**Comment by MatthewZMD** [comment]
Patch merged in 95b27717eefefffab270fb6b906eeda44cadbf8c. Thank you @nicola-leoni
